forge-game/src/main/java/forge/game/staticability/StaticAbilityDisableTriggers.java

```diff
@@ -0,0 +1,118 @@
+        // CR 603.2e
+        if (stAb.hasParam("ValidCard") && regtrig.getSpawningAbility() != null) {
+            return false;
+        }
+
+        if (!stAb.matchesValidParam("ValidCard", regtrig.getHostCard())) {
+            return false;
+        }
+
+        if (stAb.hasParam("ValidMode")) {
+            if (!ArrayUtils.contains(stAb.getParam("ValidMode").split(","), regtrig.getMode().toString())) {
+                return false;
+            }
+        }
+
+        if (triggerType.equals(TriggerType.ChangesZone)) {
+            // Cause of the trigger – the card changing zones
+            if (!stAb.matchesValidParam("ValidCause", runParams.get(AbilityKey.Card))) {
+                return false;
+            }
+            if (!stAb.matchesValidParam("Destination", runParams.get(AbilityKey.Destination))) {
+                return false;
+            }
+            if (!stAb.matchesValidParam("Origin", runParams.get(AbilityKey.Origin))) {
+                return false;
+            }
+            if ("Graveyard".equals(runParams.get(AbilityKey.Destination))
+                    && "Battlefield".equals(runParams.get(AbilityKey.Origin))) {
+                // Allow triggered ability of a dying creature that triggers
+                // only when that creature is put into a graveyard from anywhere
+                if ("Card.Self".equals(regtrig.getParam("ValidCard"))
+                        && (!regtrig.hasParam("Origin") || "Any".equals(regtrig.getParam("Origin")))) {
+                    return false;
+                }
+            }
+        } else if (triggerType.equals(TriggerType.ChangesZoneAll)) {
+            final String origin = stAb.getParam("Origin");
+            final String destination = stAb.getParam("Destination");
+            final CardZoneTable table = (CardZoneTable) runParams.get(AbilityKey.Cards);
+            CardZoneTable filtered = new CardZoneTable();
+            boolean possiblyDisabled = false;
+
+            // purge all forbidden causes from table
+            for (Cell<ZoneType, ZoneType, CardCollection> cell : table.cellSet()) {
+                CardCollection changers = cell.getValue();
+                if ((origin == null || cell.getRowKey() == ZoneType.valueOf(origin)) &&
+                (destination == null || cell.getColumnKey() == ZoneType.valueOf(destination))) {
+                    changers = CardLists.filter(changers, Predicates.not(CardPredicates.restriction(stAb.getParam("ValidCause").split(","), stAb.getHostCard().getController(), stAb.getHostCard(), stAb)));
+                    // static will match some of the causes
+                    if (changers.size() < cell.getValue().size()) {
+                        possiblyDisabled = true;
+                    }
+                }
+                filtered.put(cell.getRowKey(), cell.getColumnKey(), changers);
```

Comment:
@Hanmac
Tricky question about interaction with Panharmonicon-stuff remains:

- Make Wizard Artifact Creature + non-Creature Artifact ETB simultaneously
- Now _Hushbringer_ says "Creatures entering...**don't cause**"
- _Naban_ says "Wizard entering...**causes**"
- Should he still make_Ingenious Artillerist_ trigger twice? "Whenever one or more artifacts enter..." (because the Wizard could not have caused it)

Might have to keep using the runParamsFiltered for Panharmonicon check? 🤔 

![grafik](https://user-images.githubusercontent.com/8506892/210995494-28ef29f6-473a-4081-897f-e9f1dbb26512.png)

Reply:
_Syr Konrad_ case works fine

Reply:
some Judges in the chat said this:
> Yes,  you get one trigger, and it will deal 2 damage. The artifact creature still entered, so the Artillerist can count it since it's triggering because the noncreature artifact entered. It doesn't just ignore it completely, so you get one trigger that deals 2 damage.

:/

but i still asked Rules Manager: https://twitter.com/Hanmac10/status/1611442093204901936

Reply:
Thanks for checking, I think it kind of makes sense

it's handled now